todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.java

```diff
@@ -16,101 +16,166 @@
         super.onResume();
-        mPresenter.subscribe();
     }
 
     @Override
     public void onPause() {
+        unbindViewModel();
         super.onPause();
-        mPresenter.unsubscribe();
     }
 
-    @Override
-    public void setPresenter(@NonNull AddEditTaskContract.Presenter presenter) {
-        mPresenter = checkNotNull(presenter);
+    private void bindViewModel() {
+        mSubscription = new CompositeSubscription();
+
+        mSubscription.add(mViewModel.getSnackbarText()
+                .subscribeOn(Schedulers.computation())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        // onNext
+                        this::showSnackbar,
+                        // onError
+                        throwable -> Log.e(TAG, "Error retrieving snackbar text", throwable)));
+
+        mSubscription.add(mViewModel.getTask()
+                .subscribeOn(Schedulers.computation())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        // onNext
+                        this::setTask,
+                        // onError
+                        throwable -> Log.e(TAG, "Error retrieving the task", throwable)));
+
+    }
+
+    private void unbindViewModel() {
+        mSubscription.unsubscribe();
+    }
+
+    private void restoreData(@Nullable Bundle bundle) {
+        if (bundle == null) {
+            return;
+        }
+        mViewModel.setRestoredState(bundle.getString(TASK_TITLE_KEY),
+                bundle.getString(TASK_DESCRIPTION_KEY));
     }
 
     @Override
-    public void onActivityCreated(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
+    public void onSaveInstanceState(Bundle outState) {
+        outState.putString(TASK_TITLE_KEY, mTitle.getText().toString());
+        outState.putString(TASK_DESCRIPTION_KEY, mDescription.getText().toString());
+        super.onSaveInstanceState(outState);
+    }
 
+    private void setupFab() {
         FloatingActionButton fab =
                 (FloatingActionButton) getActivity().findViewById(R.id.fab_edit_task_done);
         fab.setImageResource(R.drawable.ic_done);
-        fab.setOnClickListener(__ -> mPresenter.saveTask(mTitle.getText().toString(),
-                mDescription.getText().toString()));
+        fab.setOnClickListener(__ -> saveTask());
     }
 
-    @Nullable
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View root = inflater.inflate(R.layout.addtask_frag, container, false);
-        mTitle = (TextView) root.findViewById(R.id.add_task_title);
-        mDescription = (TextView) root.findViewById(R.id.add_task_description);
-        setHasOptionsMenu(true);
-        return root;
+    private void saveTask() {
+        mSubscription.add(mViewModel.saveTask(mTitle.getText().toString(),
```

Comment:
There are two things happening here. How about saveTask() on one line and the sub on the other, explaining what they each do?